sdk/storage/azblob/container/client.go

```diff
@@ -329,3 +329,40 @@ func (c *Client) GetSASURL(permissions sas.ContainerPermissions, expiry time.Tim
 
 	return endpoint, nil
 }
+
+// NewBatchBuilder creates an instance of BatchBuilder with the specified values.
+//   - cred - an Azure AD credential, typically obtained via the azidentity module
+//   - options - client options; pass nil to accept the default values
+func (c *Client) NewBatchBuilder(cred azcore.TokenCredential, options *ClientOptions) *BatchBuilder {
+	return nil
+}
+
+// NewBatchBuilderWithSharedKeyCredential creates an instance of BatchBuilder with the specified values.
+//   - cred - a SharedKeyCredential created with the matching container's storage account and access key
+//   - options - client options; pass nil to accept the default values
+func (c *Client) NewBatchBuilderWithSharedKeyCredential(cred *SharedKeyCredential, options *ClientOptions) *BatchBuilder {
+	return nil
+}
+
+// NewBatchBuilderWithNoCredential creates an instance of BatchBuilder with the specified values.
+// This is used to perform batch operations anonymously or with a shared access signature (SAS) token.
```

Comment:
I don't think tis works "anonymously"; I think it has to be SAS.

Reply:
If public access is enabled on the storage account, then I think that the sub-requests won't need to have the authorization header. This part we can double check while implementing this.

Reply:
According to this https://learn.microsoft.com/en-us/rest/api/storageservices/create-container?tabs=azure-ad, public access means public READ access and so delete is not allowed and I doubt set tier is allowed as well.

Reply:
Done. Renamed the method to `NewBatchBuilderWithSAS`.

Reply:
This name implies that the SAS is passed to the ctor and never passed anywhere else. Actually, how would a SAS be passed? A SAS for each item? If so, how? If the SAS is the same for all items, then it needs to be passed to this ctor as a parameter. Would it be a string or a structure with fields set?

Reply:
The method `NewBatchBuilderWithSAS` does not take any parameters. It gets the SAS from the client's endpoint. Here the customer has to make sure that the client is created using SAS (added this as comment). We can rename this method to `NewBatchBuilderWithNoCredential` if we don't want to pass the SAS as a parameter.

If we do want to pass SAS, we can pass it to the ctor as a parameter and it will be of type string since it can be account, service or user delegation SAS.

Reply:
If you do it this way, there are some inconsistencies. First, all the other BB ctors take creds used for the items and this ctor would not. Second, when this ctor is called, we don't know if the client was created with SAS or not so we can't guarantee success. We could rename this ctor to emphasise the inconsistency something like NewBuilderUsingClientSAS (I don't like this but we can think about it more) and have this ctor return a (BB, error) with a non-nil error if the client doesn't have a SAS.

Reply:
Or, maybe we want to be consistent and let the ctor take SAS parameters which can be different than the SAS on the client?

Reply:
I like the second option where we can take SAS as a parameter which will be of `string` type.

Reply:
What about `sas.QueryParameters` instead of `string`?  Would that be better?

Reply:
Yes, I think this is good

Reply:
Added `sas.QueryParameters` as a parameter to the ctor `NewBatchBuilderWithSAS`.